server/src/graql/executor/ComputeExecutor.java

```diff
@@ -687,24 +686,19 @@ private VertexProgram initStatisticsVertexProgram(GraqlCompute query, Set<LabelI
      * @return true if they exist, false if they don't
      */
     private boolean targetContainsInstance(GraqlCompute.Statistics.Value query) {
-        for (Label attributeType : targetTypeLabels(query)) {
-            for (Label type : scopeTypeLabels(query)) {
-                Boolean patternExist = tx.stream(Graql.match(
-                        Graql.var("x").has(attributeType.getValue(), Graql.var()),
-                        Graql.var("x").isa(Graql.type(type.getValue()))
-                ), false).iterator().hasNext();
-                if (patternExist) return true;
-            }
-        }
-        return false;
-        //TODO: should use the following ask query when ask query is even lazier
-//        List<Pattern> checkResourceTypes = statisticsResourceTypes.stream()
-//                .map(type -> var("x").has(type, var())).collect(Collectors.toList());
-//        List<Pattern> checkSubtypes = inTypes.stream()
-//                .map(type -> var("x").isa(Graql.label(type))).collect(Collectors.toList());
-//
-//        return tx.get().graql().infer(false)
-//                .match(or(checkResourceTypes), or(checkSubtypes)).get().aggregate(ask()).execute();
+        Set<Label> targetLabels = targetTypeLabels(query);
+        ImmutableSet<Label> scopeLabels = scopeTypeLabels(query);
+        BiFunction<Label, Label, Pattern> patternFunction = (attributeType, type) -> Graql.and(
+                Graql.var("x").has(attributeType.getValue(), Graql.var()),
+                Graql.var("x").isa(Graql.type(type.getValue()))
+        );
```

Comment:
why is this one defined separately if it's used only once?

Reply:
yes and to avoid a cast to Pattern